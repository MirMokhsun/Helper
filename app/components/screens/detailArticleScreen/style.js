import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#F5F5F7',
    },
    title: {
        padding: 15,
        fontSize: 16,
        fontFamily: 'Roboto-Bold',
        color: '#000756'
    },
    text: {
        paddingHorizontal: 15,
        paddingBottom: 20,
        fontSize: 16,
        fontFamily: 'Roboto-Light',
        lineHeight: 21,
        color: 'rgba(0, 7, 86, 0.87)'        
    },
});