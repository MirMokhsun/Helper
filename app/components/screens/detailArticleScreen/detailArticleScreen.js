/* eslint-disable react/destructuring-assignment */
import React, { Component } from 'react';
import { View, Text, Platform, ScrollView, I18nManager } from 'react-native';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { styles } from './style';
import { dictionary } from '../../../common/faq/dictionary';
import { faq } from '../../../common/faq/faq';
import { localization } from '../../../common/localization';
import Utils from '../../../api/utils/utils';

const isIos = Utils.isIOS;
const {isRTL} = I18nManager;
const {setTextAlign} = Utils

export class DetailArticleScreen extends Component {
    static navigationOptions = ({ navigation }) => {
        return {
            headerTintColor: '#000756',
            headerBackTitle: null,
            title: navigation.state.params ? navigation.state.params.title ? navigation.state.params.title : 'Help' : 'Help',
            headerTitleStyle: {
                color: "#000756",
                alignSelf: 'center',
                width: '100%',
                height: '100%',
                textAlign: "center",
                fontSize: 20,
                lineHeight: 40,
                ...Platform.select({
                    ios: {
                        paddingRight: 0,
                    },
                    android: {
                        paddingRight: isRTL ? 0 : 30,
                        paddingLeft: isRTL ? 30 : 0,
                    },
                }),
                fontFamily: 'Roboto-Regular',
                backgroundColor: "#F5F5F7",
            },
            headerStyle: {
                backgroundColor: "#F5F5F7",
                paddingTop: isIos ? 20 : 10,
                borderBottomColor: "#F5F5F7",
                shadowOpacity: 0,
                elevation: 0,
            },
            headerRight: (<View />),
        }
    };

    constructor(props) {
        super(props);
        const { language } = props;
        this.state = {
            currentTextAlign: setTextAlign(language)
        }
    }

    componentWillMount() {
        const { navigation, language } = this.props;
        navigation.setParams({ title: localization[language]['Help'] });
    }


    componentDidUpdate({ language: prevLanguage }) {
        const { language } = this.props;
        if (language !== prevLanguage) {
            this.setState({ currentTextAlign: setTextAlign(language) });
        }
    }

    render() {
        const {
            index,
            isFaq, // нужен для того что бы выбирать между словариком и faq
            language
        } = this.props.navigation.state.params;
            const currentlanguage = isFaq ? faq[language][index] : dictionary[language][index];
            const { currentTextAlign } = this.state;
        return (
            <View testID='welcome' style={styles.container}>
                <ScrollView>
                    <Text style={[styles.title, { textAlign: currentTextAlign }]} >{currentlanguage['title']}</Text>
                    <Text style={[styles.text, { textAlign: currentTextAlign }]}>{currentlanguage['text']}</Text>
                </ScrollView >
            </View>
        );
    }
}

const mapStateToProps = state => ({
    language: state.currentLanguage.language,
});

const mapDispatchToProps = dispatch => bindActionCreators({
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(DetailArticleScreen);
