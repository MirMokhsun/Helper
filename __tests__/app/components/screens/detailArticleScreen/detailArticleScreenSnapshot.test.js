import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { DetailArticleScreen } from '../../../../../app/components/screens/detailArticleScreen/detailArticleScreen';

let mock = {
    "state": {
        "params": {
            "language": 'ru',
            "isFaq": true,
            "index": 5,
        }
    },
    setParams: () => { }
}

it('create snapshot DetailArticleScreen', () => {
    const renderer = new ShallowRenderer()
    let detailArticleScreen = renderer.render(<DetailArticleScreen navigation={mock} language={'ru'} />);
    expect(detailArticleScreen).toMatchSnapshot();
});