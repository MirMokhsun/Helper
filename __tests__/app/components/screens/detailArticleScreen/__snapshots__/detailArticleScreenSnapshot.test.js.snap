// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot DetailArticleScreen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5F5F7",
      "flex": 1,
    }
  }
  testID="welcome"
>
  <ScrollViewMock>
    <Text
      style={
        Array [
          Object {
            "color": "#000756",
            "fontFamily": "Roboto-Bold",
            "fontSize": 16,
            "padding": 15,
          },
          Object {
            "textAlign": "left",
          },
        ]
      }
    >
      География Forex
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "rgba(0, 7, 86, 0.87)",
            "fontFamily": "Roboto-Light",
            "fontSize": 16,
            "lineHeight": 21,
            "paddingBottom": 20,
            "paddingHorizontal": 15,
          },
          Object {
            "textAlign": "left",
          },
        ]
      }
    >
      Рабочими площадками банковских операций рынка Forex являются взаимодействующие между собой с помощью новейших информационных технологий региональных валютные рынки. Основными игроками являются Новозеландский и Австралийский валютные рынки (Веллингтон, Сидней), Азиатский (Токио, Сингапур, Гонконг), Европейский (Лондон, Франкфурт-на-Майне, Цюрих) и Американский (Нью-Йорк, Чикаго, Лос-Анджелес) рынки.Благодаря тому, что рынки расположены в разных частях планеты и принятой во всем мире системе часовых поясов, рынок Forex функционирует круглосуточно. Лишь один раз в неделю, с 22.00 в пятницу по Гринвичу (GMT+3 по отношению к московскому), времени закрытия американских бирж, до 21.00 воскресенья, время открытия валютных торгов в Новой Зеландии, рынок уходит на уик-энд. Также биржи не работают в периоды больших праздников.
    </Text>
  </ScrollViewMock>
</View>
`;
